### PR TITLE
Feature/jr2 116 job search service

### PR DESCRIPTION
**Pull request by birom**

### CODE REVIEW

File: src/main/java/ch/admin/seco/service/job/domain/valueobject/Occupation.java [hunk 1]
```diff
@@ -126,11 +144,12 @@ public boolean equals(Object o) {
...
     public String toString() {
         return "Occupation{" +
-            "code='" + getCode() + "'" +
-            ", qualificationLevel='" + getQualificationLevel() + "'" +
-            ", relation='" + getRelation() + "'" +
-            ", experienceInYears='" + getExperienceInYears() + "'" +
-            ", educationLevel='" + getEducationLevel() + "'" +
-            "}";
+                "code='" + getCode() + "'" +
+                "classificationCode='" + getClassificationCode() + "'" +
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/domain/valueobject/Occupation.java]: prefix the label with a comma, as it's not the first value
`", classificationCode='"`

File: src/main/java/ch/admin/seco/service/job/service/JobService.java [hunk 2]
```diff
@@ -47,9 +48,9 @@
     /**
      * Search for the job corresponding to the query.
      *
-     * @param query    the query of the search
+     * @param request    the request of the search
      * @param pageable the pagination information
      * @return the list of entities
      */
-    Page<Job> search(String query, Pageable pageable);
+    Page<Job> search(JobSearchRequestDTO request, Pageable pageable);
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/service/JobService.java]: `request` is a very generic name for a parameter. better use `jobSearchRequest`

File: src/main/java/ch/admin/seco/service/job/domain/valueobject/Occupation.java [hunk 3]
```diff
@@ -22,6 +22,11 @@
     @Column(name = "code", length = 6, nullable = false)
     private String code;
 
+    @NotNull
+    @Size(max = 3)
+    @Column(name = "classification_code", length = 3, nullable = false)
+    private String classificationCode;
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/domain/valueobject/Occupation.java]: should we use `int` or `String` for occupation-`code` and `classificationCode`?

Here we defined String - in the JobSearchRequestDTO we defined int's

File: src/main/java/ch/admin/seco/service/job/service/impl/JobServiceImpl.java [hunk 4]
```diff
@@ -91,15 +102,59 @@ public void delete(UUID id) {
...
+     * @param request    the request of the search
      * @param pageable the pagination information
      * @return the list of entities
      */
     @Override
     @Transactional(readOnly = true)
-    public Page<Job> search(String query, Pageable pageable) {
-        log.debug("Request to search for a page of Jobs for query {}", query);
-        Page<Job> result = jobSearchRepository.search(queryStringQuery(query), pageable);
+    public Page<Job> search(JobSearchRequestDTO request, Pageable pageable) {
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/service/impl/JobServiceImpl.java]: rename parameter name

File: src/main/java/ch/admin/seco/service/job/service/impl/JobServiceImpl.java [hunk 5]
```diff
@@ -91,15 +102,59 @@ public void delete(UUID id) {
...
+            if (!isEmpty(request.getOccupations())) {
+                ((BoolQueryBuilder) query).should(termsQuery("occupations.code", request.getOccupations()));
+            }
+            if (!isEmpty(request.getClassifications())) {
+                ((BoolQueryBuilder) query).should(termsQuery("occupations.classificationCode", request.getClassifications()));
+            }
+        }
+
+
+        final QueryBuilder filter;
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/service/impl/JobServiceImpl.java]: why do we use filter here? what is the consideration? queries will affect the calculated "score", filter won't

File: src/main/java/ch/admin/seco/service/job/web/rest/JobResource.java [hunk 6]
```diff
@@ -135,19 +135,19 @@ public JobResource(JobService jobService) {
...
      * @param pageable the pagination information
      * @return the result of the search
      */
     @GetMapping("/_search/jobs")
     @Timed
-    public ResponseEntity<List<Job>> searchJobs(@RequestParam String query, @ApiParam Pageable pageable) {
-        log.debug("REST request to search for a page of Jobs for query {}", query);
-        Page<Job> page = jobService.search(query, pageable);
-        HttpHeaders headers = PaginationUtil.generateSearchPaginationHttpHeaders(query, page, "/api/_search/jobs");
+    public ResponseEntity<List<Job>> searchJobs(@ApiParam JobSearchRequestDTO request, @ApiParam Pageable pageable) {
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/web/rest/JobResource.java]: rename parameter name

File: src/main/java/ch/admin/seco/service/job/service/impl/JobServiceImpl.java (hunk 5 above)
Comment (birom) [inline src/main/java/ch/admin/seco/service/job/service/impl/JobServiceImpl.java]: Thanks for the review! I've fixed the findings in my last commit.

**Pull request merged**